designs/UI/myJS.js

```diff
@@ -24,15 +51,42 @@ $( document ).ready(function() {
 
     function addNewIng(){
         $("#ingredients input[name=last-ingredient]").on("focus", function(){
+            let prevIngredient = $(this).parent().parent().find(`.ingredient${ing_num-1}`);
 
-            $("#ingredients #last-ingredient-list").before('<li><input class="ingredient' + ing_num + '" type="text" name="' + name+ing_num + '"></li>');
-            $("#ingredients input[name=" + name+ing_num + "]").focus();
-            ing_num += 1;
+            if (checkInputField(prevIngredient.val()) == true){
+                $("#ingredients #last-ingredient-list").before('<li><input class="validate ingredient' + ing_num + '" type="text" name="' + name+ing_num + '"><button type="button" class="close del-ingredient" aria-label="Delete Ingredient"><span aria-hidden="true">&times;</span></button></li>');
+                $("#ingredients input[name=" + name+ing_num + "]").focus();
+                ing_num += 1;
+            }
 
         });
     }
 
-    // When a user clicks on the edit btn
+    function deleteIngredient(ingredient){
+        del_ing_num = ingredient.find("input").attr('class').match(/\d+/)[0];
+        del_ing_num=parseInt(del_ing_num);
+
+        update_ing_num = del_ing_num;
+
+
+        ingredient.remove();
+        console.log(typeof(del_ing_num));
```

Comment:
Why is this stil here?

Reply:
I forgot to remove it, sorry.

Reply:
Also, remove the unnecessary lines. Good implementation of JS here